docs/how-to-write-function.md

```diff
@@ -4,104 +4,150 @@ This document sets out some guidelines for writing functions for Scalar DL.
 
-`JsonObject` for a function argument is a JSON object and a runtime argument for the function specified by the requester. The argument is not digitally signed as opposed to the contract argument so that it can be used to pass data that is stored in the database but it might be deleted at some later point for some reason.
+### About the `invoke` arguments
 
-`contractArgument` and `contractProperties` are the corresponding contract's argument and properties. See [the contract guide](how-to-write-contract.md) to understand what they are.
+Similar to a Contract using `Ledger` object to manage assets, a Function uses `Database` object to manage records of the underlying database. Note that `Database` implements [Scalar DB](https://github.com/scalar-labs/scalardb) interface so that you can do the CRUD operations base on [the data model](https://github.com/scalar-labs/scalardb/blob/master/docs/design.md#data-model) of Scalar DB. 
+
+A `functionArgument` is a runtime argument for the Function specified by the requester. The argument is not digitally signed as opposed to the contract argument so that it can be used to pass data that is stored in the database but it might be deleted at some later point for some reason.
+
+`contractArgument` and `contractProperties` are the corresponding contract's argument and properties. See [How to Write a Contract](how-to-write-contract.md) to understand what they are.
 
 ### How to use Functions
 
-The Function feature is enabled by default; thus, nothing needs to be configured in Ledger except for the following things.
-If you want to disable the feature, please set `scalar.dl.ledger.function.enabled` to `false` in the properties of Ledger.
+The Function feature is enabled by default; thus, nothing needs to be configured in Ledger except for the following things. If you want to disable the feature, please set `scalar.dl.ledger.function.enabled` to `false` in the properties of Ledger.
 
 #### Add an application-specific schema
 
-Since Functions can read and write arbitrary records through the Scalar DB CRUD interface, Scalar DL can't define the database schema for the Function by itself.
-It is the applications' owner's responsibility to define such schema and apply it to the database by themselves or asking system admins to do it depending on who owns and manages the database.
+Since Functions can read and write arbitrary records through the Scalar DB CRUD interface, Scalar DL can't define the database schema for the Function by itself. It is the applications' owner's responsibility to define such schema and apply it to the database by themselves or asking system admins to do it depending on who owns and manages the database.
```

Comment:
```suggestion
Since Functions can read and write arbitrary records through the Scalar DB CRUD interface, Scalar DL can't define the [database schema](https://github.com/scalar-labs/scalardb/blob/master/docs/schema.md) for the Function by itself. It is the applications' owner's responsibility to define such schema and apply it to the database by themselves or asking system admins to do it depending on who owns and manages the database.
```
I think it would be helpful to have a link of `Database schema in Scalar DB` here since that document explains schema of Scalar DB (it also has a link to document of Schema Loader).

Reply:
I updated the section based on your feedback in [8568d33](https://github.com/scalar-labs/scalardl/pull/29/commits/8568d33a848f9be897858373db56967b3ddca551).